doc/index.md

```diff
@@ -1337,68 +1353,125 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
+  - `.mask` (*boolean*): Set and restore umask when binding UNIX domain socket.
+  - `.unlink` (*boolean*): `true` means unlink socket path before binding.
+  - `.reuseaddr` (*boolean*): Turn on `SO_REUSEADDR` flag.
+  - `.reuseport` (*boolean*): Turn on `SO_REUSEPORT` flag.
 
 
 ### `server:onerror(new_handler)` <!-- --> {#http.server:onerror}
 
+If called with parameters, the function replaces the current error handler function with `new_handler`  and returns a reference to the old function. Calling the function with no parameters returns the current error handler. The default handler throws an error. The `onerror` function for the server can be set during instantiation through the `options` table passed to the [*server.listen(options)*](#server.listen) function.
+
 
 ### `server:listen(timeout)` <!-- --> {#http.server:listen}
 
+Initializes the server socket and if required, resolves DNS. *server:listen* is required if [*localname*](#http.server:localname) is called before [*step*](#http.server:step) or [*loop*](#http.server:loop). On error, returns `nil`, an error message and an error number.
+
 
 ### `server:localname()` <!-- --> {#http.server:localname}
 
+Returns the connection information for the local socket. Returns address family, IP address and port for an external socket. For Unix domain sockets, the function returns AF_UNIX and the path. If the connection object is not connected, returns AF_UNSPEC (0). On error, returns `nil` an error message and an error number.
+
 
 ### `server:pause()` <!-- --> {#http.server:pause}
 
-Cause the server loop to stop processing new clients until [`:resume`](#http.server:resume) is called.
+Cause the server loop to stop processing new clients until [*resume*](#http.server:resume) is called. Existing client connections will run until closed. 
 
 
 ### `server:resume()` <!-- --> {#http.server:resume}
 
+Resumes a [*paused*](#http.server:pause) `server` and processes new client requests. 
+
 
 ### `server:close()` <!-- --> {#http.server:close}
 
+Shutdown the server and close the socket. A closed server cannot be reused.
+
 
 ### `server:pollfd()` <!-- --> {#http.server:pollfd}
 
+Returns a file descriptor (as an integer) or `nil`. The file descriptor can be passed to a system API like select or kqueue to wait on anything this server object wants to do. [*pollfd*] is used for integrating with other main loops, and should be used in combination with [*events*](#http.server:events) and [*timeout*](#http.server:timeout). 
+
 
 ### `server:events()` <!-- --> {#http.server:events}
 
+Returns a string representing the poll-mask that should be used to poll the object. If the object polls on `POLLIN`, *events* will return `"r"`. If the object polls on `POLLOUT` *events* returns `"w"`
+
+Note that the system may  also return `POLLIN|POLLOUT` as `"rw"` and a telent service may use `'rp'` for `POLLIN|POLLPRI`. These features are not commonly implemented in TCP communications. 
```

Comment:
two spaces between "may" and "also".

But I don't really like how this section is worded. Should be more like:

Returns a string indicating the type of events the object is waiting on: the string will contain `"r"` if it wants to be `:step()` d when `:pollfd()` has had `POLLIN` indicated; `"w"` for `POLLOUT` or `"p"` for `POLLPRI`. This interface is compatible with cqueues.